### PR TITLE
Use context uri in expanded resources

### PR DESCRIPTION
**Pull request by brinxmat**
- Removes the @context at the latest possible moment to avoid disrupting anything in any other process
- Does this in a rather ignorant way, but this is unavoidable at this point, alas.

### CODE REVIEW

File: expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java [hunk 1]
```diff
@@ -118,6 +128,15 @@ public ExpandedDataEntry expandEntry(Entity dataEntry) throws JsonProcessingExce
         throw new UnsupportedOperationException(UNSUPPORTED_TYPE + dataEntry.getClass().getSimpleName());
     }
 
+    private ObjectNode replaceInlineContextWithUriContext(String resourceWithNviValues) {
+        var objectNode =  attempt(() -> (ObjectNode) objectMapper.readTree(resourceWithNviValues)).orElseThrow();
```
Comment (sondrev) [inline expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java:132]: double space between = and attempt

File: expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java [hunk 2]
```diff
@@ -118,6 +128,15 @@ public ExpandedDataEntry expandEntry(Entity dataEntry) throws JsonProcessingExce
...
+        if (!objectNode.at(CONTEXT_NODE_SELECTOR).isMissingNode()) {
+            objectNode.remove(CONTEXT_FIELD_PROPERTY_NAME);
+            objectNode.put(CONTEXT_FIELD_PROPERTY_NAME, CONTEXT_URI);
```
Comment (sondrev) [inline expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java:135]: .put replaced content, so it should be unecasery to remove() first

File: expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java [hunk 3]
```diff
@@ -197,6 +198,21 @@ void shouldReturnFramedIndexDocumentFromResource(Class<?> instanceType)
...
+
+        Publication publication = randomPublication(AcademicArticle.class)
+                                      .copy()
+                                      .withEntityDescription(new EntityDescription())
+                                      .build();
+
+        Resource resourceUpdate = Resource.fromPublication(publication);
+        ExpandedResource indexDoc = (ExpandedResource) expansionService.expandEntry(resourceUpdate);
+        assertThat(indexDoc.getAllFields().get("@context"),
+                   is(equalTo("https://api.dev.nva.aws.unit.no/publication/context")));
```
Comment (monaullah) [inline expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java:213]: Read api host from environment?

**Review by sondrev — APPROVED** [review]

File: expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java [hunk 4]
```diff
@@ -118,6 +128,15 @@ public ExpandedDataEntry expandEntry(Entity dataEntry) throws JsonProcessingExce
         throw new UnsupportedOperationException(UNSUPPORTED_TYPE + dataEntry.getClass().getSimpleName());
     }
 
+    private ObjectNode replaceInlineContextWithUriContext(String resourceWithNviValues) {
+        var objectNode =  attempt(() -> (ObjectNode) objectMapper.readTree(resourceWithNviValues)).orElseThrow();
+        if (!objectNode.at(CONTEXT_NODE_SELECTOR).isMissingNode()) {
```
Comment (truhacevkir) [inline expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java:133]: if (hasContext(objectNode)) or something similar in conditional

File: expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java [hunk 5]
```diff
@@ -197,6 +198,21 @@ void shouldReturnFramedIndexDocumentFromResource(Class<?> instanceType)
...
+        throws JsonProcessingException, NotFoundException {
+
+        Publication publication = randomPublication(AcademicArticle.class)
+                                      .copy()
+                                      .withEntityDescription(new EntityDescription())
+                                      .build();
+
+        Resource resourceUpdate = Resource.fromPublication(publication);
+        ExpandedResource indexDoc = (ExpandedResource) expansionService.expandEntry(resourceUpdate);
+        assertThat(indexDoc.getAllFields().get("@context"),
```
Comment (truhacevkir) [inline expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java:212]: vars

File: expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java (hunk 3 above)
Comment (brinxmat) [reply expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java:213]: I'm not keen on reading from env in tests…we don't test the expectation, just the code. If someone mutates the string, it won't fail

File: expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java (hunk 2 above)
Comment (brinxmat) [reply expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java:135]: Ha, I didn't think it did, but apparently so!

File: expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java [hunk 6]
```diff
@@ -129,14 +129,17 @@ public ExpandedDataEntry expandEntry(Entity dataEntry) throws JsonProcessingExce
     }
 
     private ObjectNode replaceInlineContextWithUriContext(String resourceWithNviValues) {
-        var objectNode =  attempt(() -> (ObjectNode) objectMapper.readTree(resourceWithNviValues)).orElseThrow();
-        if (!objectNode.at(CONTEXT_NODE_SELECTOR).isMissingNode()) {
-            objectNode.remove(CONTEXT_FIELD_PROPERTY_NAME);
+        var objectNode = attempt(() -> (ObjectNode) objectMapper.readTree(resourceWithNviValues)).orElseThrow();
+        if (hasContextNode(objectNode)) {
```
Comment (sondrev) [inline expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java:133]: do we even need to check? Cant we just always put the context uri?

**Review by monaullah — APPROVED** [review]

**Review by monaullah — APPROVED** [review]

**Review by truhacevkir — APPROVED** [review]

File: expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java (hunk 6 above)
Comment (brinxmat) [reply expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java:133]: In tests, this isn't always the case alas

**Pull request merged**